src/ldeboot.c

```diff
@@ -200,7 +200,12 @@ int main(int argc, char *argv[]) {
     argv[0] = filetorun;
   } else {
     /* copy up to and including the final "/" in the path */
+#ifndef MAIKO_OS_HAIKU 
```

Comment:
`strncpy(dst, src, ...)` is documented as returning `dst`, while `stpncpy(dst, src, ...)` is documented as returning a pointer to the terminating NUL (if there is one) of `dst`.  They are NOT interchangeable.

Reply:
Haiku for stpncpy in hrev57422